certmanager/ca.go

```diff
@@ -50,22 +63,36 @@ func NewSelfSignedIstioCA() *IstioCA {
 		IsSelfSigned: true,
 	}
 	pemCert, pemKey := GenCert(options)
-	cert, key := parsePemEncodedCertificateAndKey(pemCert, pemKey)
 
-	return NewIstioCA(cert, key)
+	opts := &IstioCAOptions{
+		CertChainBytes:   pemCert,
```

Comment:
Done. We are on longer sending root cert in the cert chain in self-signed case. 

In non-self-signed case we have no control over what's in the chain file. So we'll blindly send over what's in the file.